back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -429,6 +432,10 @@ def compute_stock_overview(base_id):
             boxes.c.box_state,
             Product.category.alias("category_id"),
             fn.TRIM(fn.LOWER(Product.name)).alias("product_name"),
+            fn.CONCAT(
+                fn.ROUND(boxes.c.measure_value, 3),
+                Case(Unit.dimension_id, ((28, "kg"), (29, "l")), ""),
```

Comment:
grouping over a conditional in a column is one of the most expensive actions one can do in SQL. I'd suggest another solution. 
Rather query both columns (measure value and dimension) and group by both columns and then do the concat.